src/token/cose/sign/sign1.rs

```diff
@@ -0,0 +1,430 @@
+            aad.lookup_aad(protected.as_ref(), unprotected.as_ref()),
+            |tosign| {
+                sign::try_sign(
+                    backend,
+                    key_provider,
+                    protected.as_ref(),
+                    unprotected.as_ref(),
+                    tosign,
+                )
+            },
+        )
+    }
+    fn try_sign_detached<
+        'a,
+        'b,
+        B: CoseSignCipher,
+        CKP: CoseKeyProvider<'a>,
+        CAP: CoseAadProvider<'b>,
+    >(
+        self,
+        backend: &mut B,
+        key_provider: &mut CKP,
+        protected: Option<Header>,
+        unprotected: Option<Header>,
+        payload: &[u8],
+        mut aad: &mut CAP,
+    ) -> Result<Self, CoseCipherError<B::Error>> {
+        let mut builder = self;
+        if let Some(protected) = &protected {
+            builder = builder.protected(protected.clone());
+        }
+        if let Some(unprotected) = &unprotected {
+            builder = builder.unprotected(unprotected.clone());
+        }
+        builder.try_create_detached_signature(
+            payload,
+            aad.lookup_aad(protected.as_ref(), unprotected.as_ref()),
+            |tosign| {
+                sign::try_sign(
+                    backend,
+                    key_provider,
+                    protected.as_ref(),
+                    unprotected.as_ref(),
+                    tosign,
+                )
+            },
+        )
+    }
+}
+
+pub trait CoseSign1Ext {
+    fn try_verify<'a, 'b, B: CoseSignCipher, CKP: CoseKeyProvider<'a>, CAP: CoseAadProvider<'b>>(
+        &self,
+        backend: &mut B,
+        key_provider: &mut CKP,
+        try_all_keys: bool,
+        aad: &mut CAP,
+    ) -> Result<(), CoseCipherError<B::Error>>;
+
+    fn try_verify_detached<
+        'a,
+        'b,
+        B: CoseSignCipher,
+        CKP: CoseKeyProvider<'a>,
+        CAP: CoseAadProvider<'b>,
+    >(
+        &self,
+        backend: &mut B,
+        key_provider: &mut CKP,
+        try_all_keys: bool,
+        payload: &[u8],
+        aad: &mut CAP,
+    ) -> Result<(), CoseCipherError<B::Error>>;
+}
+
+impl CoseSign1Ext for CoseSign1 {
+    fn try_verify<'a, 'b, B: CoseSignCipher, CKP: CoseKeyProvider<'a>, CAP: CoseAadProvider<'b>>(
+        &self,
+        backend: &mut B,
+        key_provider: &mut CKP,
+        try_all_keys: bool,
+        aad: &mut CAP,
+    ) -> Result<(), CoseCipherError<B::Error>> {
+        self.verify_signature(
+            aad.lookup_aad(Some(&self.protected.header), Some(&self.unprotected)),
+            |signature, toverify| {
+                sign::try_verify(
+                    backend,
+                    key_provider,
+                    &self.protected.header,
+                    &self.unprotected,
+                    try_all_keys,
+                    signature,
+                    toverify,
+                )
+            },
+        )?;
+
+        Ok(())
```

Comment:
Can't we return `verify_signature` directly here?
```suggestion
        )
```